pyresample/geometry.py

```diff
@@ -693,16 +694,29 @@ def geocentric_resolution(self, ellps='WGS84', radius=None, nadir_factor=2):
         if self.ndim == 1:
             raise RuntimeError("Can't confidently determine geocentric "
                                "resolution for 1D swath.")
-        rows = self.shape[0]
+
+        if isinstance(self.lons, xr.DataArray):
+            rows = self.lons['y'].shape[0]
+        else:
+            # Data have no information on dimensions, so we assume first dimension (the rows) is the y-axis:
+            logger.warning('As Numpy data arrays carry no information on the data layout we here ' +
+                           'assume the first dimension (the rows) is the y-axis (the satellite scans)')
+            rows = self.shape[0]
```

Comment:
So users get a warning for *every* execution of this method? No, we can't do this.

Reply:
`warnings.warn` would in theory only issue a warning on the first execution.

Reply:
Lol right, sure, but the overall point still stands. I would still prefer that users receive no warning for a usage that has existed for years and years and is perfectly fine in 99.9999% of cases that we run into.

My preference is still that the Satpy reader reorient data, but I also have no experience with that instrument.

Reply:
Ok I take back the "years and years" thing, I thought this was a different method being modified...but still.

Reply:
I agree that in this particular case, there should be no warning at all.

Reply:
Ha, very good, I did trigger a reaction here! I was pretty sure you wouldn't like this when I wrote it, but wanted to discuss it. Done now. So sorry, I take it back again.

Reply:
Ha, very good, I did trigger a reaction here! I was pretty sure you wouldn't like this when I wrote it, but wanted to discuss it. Done now. So sorry, I take it back again.